### PR TITLE
refactor: remove goto statements from replica.cc

### PR DESCRIPTION
**Pull request by kostasrim**
* remove goto statements from replica.cc

### CODE REVIEW

File: src/server/replica.cc [hunk 1]
```diff
@@ -1001,9 +1001,14 @@ error_code Replica::ParseReplicationHeader(base::IoBuf* io_buf, PSyncResponse* d
   std::string_view header;
   bool valid = false;
 
+  auto bad_header = [str]() {
+    LOG(ERROR) << "Bad replication header: " << str;
```
Comment (dranikpg) [inline src/server/replica.cc:1005]: Just an idea, this function is called only once, maybe it'd be easier to handle the illegal opcode on the caller site?

Comment (kostasrim) [reply src/server/replica.cc:1005]: I am slightly confused. Are you referring to `bad_header` ? This is called in a few places?

Comment (dranikpg) [reply src/server/replica.cc:1005]: No no, ParseReplicationHeader

Comment (kostasrim) [reply src/server/replica.cc:1005]: Uh I see. The thing is that we `LOG` the bad header before we exit. That means that we would need to return a string_view with the bad header to LOG it on the calling site. However this function needs to return `error_code` for two reasons: a) calling conversion consistency: `RETURN_ON_ERR(ParseReplicationHeader(&io_buf, &repl_header));` b) we return error_code on `ReadLine` above

**Pull request merged**